tests/web/rpc/v1/test_storage_routing.py

```diff
@@ -173,6 +199,41 @@ def test_merge_query_settings() -> None:
     assert routing_decision.clickhouse_settings == {"some_setting": "some_value"}
 
 
+@pytest.mark.redis_db
+def test_routing_strategy_selects_tier_1_if_highest_accuracy_mode() -> None:
+    ts = Timestamp()
+    ts.GetCurrentTime()
+    tstart = Timestamp(seconds=ts.seconds - 3600)
+    # don't set downsampled_storage_config
+    in_msg = TimeSeriesRequest(
+        meta=RequestMeta(
+            request_id=RANDOM_REQUEST_ID,
+            project_ids=[1, 2, 3],
+            organization_id=1,
+            cogs_category="something",
+            referrer="something",
+            start_timestamp=tstart,
+            end_timestamp=ts,
+            trace_item_type=TraceItemType.TRACE_ITEM_TYPE_SPAN,
+        ),
+        granularity_secs=60,
+    )
+    routing_context = deepcopy(ROUTING_CONTEXT)
+    routing_context.in_msg = in_msg
+    routing_decision = RoutingStrategyHighestAccuracy().get_routing_decision(
+        routing_context
+    )
+    assert routing_decision.tier == Tier.TIER_8
+
+    in_msg.meta.downsampled_storage_config.mode = (
+        DownsampledStorageConfig.MODE_HIGHEST_ACCURACY
+    )
+    routing_decision = RoutingStrategyHighestAccuracy().get_routing_decision(
+        routing_context
+    )
+    assert routing_decision.tier == Tier.TIER_1
+
+
```

Comment:
Does it make sense to add a test that verifies at least one non highest accuracy enum like `NORMAL`? For example, verifying that the tier is `TIER_8` if storage config mode is `NORMAL`?

Reply:
not really. That functionality is not built in to the base class. The individual strategies have to decide how to use normal mode